### ISSUE TITLE
Can regular expression be used in Conditional Expressions? Any string functions available?

### ISSUE THREAD

**Issue by meicale** [casey/just#959]
I want to test whether a sub-string (such as "GPU") is contained in a parameter or values.
I tried if "GPUS" == *"GPU"*, which can work in the shell and return the 'true' value.

**Comment by casey** [comment]
This isn't currently possible, but I think it would be a useful feature. I would propose adding it as a new conditional form, probably something like:

```
x := if foo ~= "bar" { a } else { b }
```

Where `"bar"` is a regex. We could also add slash-delimited regex literals, `/…/`, but I can't think of any strong advantages over just reusing string literals for this specific case, and adding regex literals if we need them later.

**Comment by meicale** [comment]
Thanks！

In my case, it is used to handle the parameters given, which may vary a little in different running.  I think it is a need, and is a common in sh or Python scripts.

**Comment by hustcer** [comment]
I need this feature, too. Thx

**Comment by casey** [comment]
Implemented in #970! I'll try to cut a release soon, but for now it's available on master on GitHub.

The operator is `=~` since that's what's used in Perl, Ruby, and other similar languages:

```rust
foo := if "hello" =~ 'hel+o' { "match" } else { "mismatch" }

bar:
  @echo {{foo}}
```

```
$ just bar
match
```

**Comment by casey** [comment]
I just published 0.10.2, which includes regular expression conditionals. Prebuilt binaries should be available soon on the releases page. Sorry for the delay! I was on vacation 🙇‍♂️

**Comment by meicale** [comment]
good job!
